mineinabyss-features/src/main/kotlin/com/mineinabyss/relics/StarCompassListener.kt

```diff
@@ -0,0 +1,51 @@
+package com.mineinabyss.relics
+
+import com.mineinabyss.components.playerData
+import com.mineinabyss.components.relics.StarCompass
+import com.mineinabyss.deeperworld.world.section.section
+import com.mineinabyss.geary.autoscan.AutoScan
+import com.mineinabyss.geary.ecs.accessors.TargetScope
+import com.mineinabyss.geary.ecs.accessors.building.get
+import com.mineinabyss.geary.ecs.api.systems.TickingSystem
+import com.mineinabyss.helpers.bossbarCompass
+import com.mineinabyss.idofront.items.editItemMeta
+import com.mineinabyss.looty.tracking.toGearyOrNull
+import net.kyori.adventure.bossbar.BossBar
+import net.kyori.adventure.text.Component
+import org.bukkit.Location
+import org.bukkit.Material
+import org.bukkit.entity.Player
+import org.bukkit.inventory.meta.CompassMeta
+import kotlin.time.Duration.Companion.seconds
+
+@AutoScan
+class StarCompassSystem : TickingSystem(interval = 0.1.seconds) {
+    private val TargetScope.player by get<Player>()
+    private val compassBar = BossBar.bossBar(Component.text(":arrow_null:"), 1.0f, BossBar.Color.WHITE, BossBar.Overlay.PROGRESS)
+
+    override fun TargetScope.tick() {
+        val compass = player.inventory.contents.firstOrNull {
```

Comment:
Do not tick the entire inventory of every player on the server for one specific item. Query for the `StarCompass` component instead. To access player you should be able to do `parent.get<Player>()` instead